### PR TITLE
chore: #42 - Add helm chart

### PR DESCRIPTION
**Pull request by sathieu**
Add basic helm chart.

Fixes: #42

### CODE REVIEW

**Comment by grzesuav** [comment]
hi @sathieu , thanks for PR, could you use instructions to docs about the chelm chart and how to use it, i.e. which values are customizable and what do they mean ?

File: helm/metacontroller/crds/metacontroller-crds-v1.yaml [hunk 1]
```diff
@@ -0,0 +1,534 @@
+
```
Comment (grzesuav) [inline helm/metacontroller/crds/metacontroller-crds-v1.yaml:1]: can't it include this file from `manifests/` directory ?

**Comment by grzesuav** [comment]
also, could you change commit type to `chore`/`bugfix` as it does not bring any code change ?

**Comment by grzesuav** [comment]
also, please rebase on latest master to fix pipelines, just merged fix

**Comment by fpetkovski** [comment]
Should we keep all deployment manifests in the `deploy` folder?
So the tree would look like
```
deploy
-- kustomize
-- helm
```

File: helm/metacontroller/crds/metacontroller-crds-v1.yaml (hunk 1 above)
Comment (sathieu) [reply helm/metacontroller/crds/metacontroller-crds-v1.yaml]: Unfortunately not (or at least I've not found a way).

**Comment by sathieu** [comment]
@grzesuav I've rebased to fix the CI, renamed `chore`, moved to `deploy/helm` and added docs. Back to you 🏸 !

File: deploy/helm/metacontroller/templates/statefulset.yaml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+      containers:
+        - name: {{ .Chart.Name }}
+          securityContext:
+            {{- toYaml .Values.securityContext | nindent 12 }}
+          image: "{{ .Values.image.repository }}:{{ .Values.image.tag | default .Chart.AppVersion }}"
+          imagePullPolicy: {{ .Values.image.pullPolicy }}
+          command:
+          - /usr/bin/metacontroller
+          args:
+          - --logtostderr
```
Comment (grzesuav) [inline deploy/helm/metacontroller/templates/statefulset.yaml:39]: Those args should be configurable and allow to set any of the metacontroller args. I recommend to do some templating there and define default set in `values`

**Review by grzesuav — APPROVED** [review]
I think that for first version it it good enough, I have just one comment with providing more configurable args options to chart to make it more usable

**Comment by grzesuav** [comment]
hi @sathieu , thanks for this, I have several improvements in mind, but do not want to keep it for too long. They can be addressed as separate PR's.

**Comment by grzesuav** [comment]
> Should we keep all deployment manifests in the `deploy` folder?
> So the tree would look like
> 
> ```
> deploy
> -- kustomize
> -- helm
> ```

good idea, will adjust manifests in next PR

**Comment by grzesuav** [comment]
Created https://github.com/metacontroller/metacontroller/issues/302 for improvements

**Pull request merged**